glean-core/rlb/src/test.rs

```diff
@@ -481,3 +483,63 @@ fn setting_source_tags_before_initialization_should_not_crash() {
 fn flipping_upload_enabled_respects_order_of_events() {
     todo!()
 }
+
+#[test]
+fn registering_pings_before_init_must_work() {
+    let _lock = lock_test();
+
+    destroy_glean(true);
+    assert!(!was_initialize_called());
+
+    // Define a fake uploader that reports back the submission headers
+    // using a crossbeam channel.
+    let (s, r) = crossbeam_channel::bounded::<String>(1);
+
+    #[derive(Debug)]
+    pub struct FakeUploader {
+        sender: crossbeam_channel::Sender<String>,
+    };
+    impl net::PingUploader for FakeUploader {
+        fn upload(
+            &self,
+            url: String,
+            _body: Vec<u8>,
+            _headers: Vec<(String, String)>,
+        ) -> net::UploadResult {
+            self.sender.send(url).unwrap();
+            net::UploadResult::HttpStatus(200)
+        }
+    }
+
+    // Create a custom ping and attempt its registration.
+    #[allow(non_upper_case_globals)]
+    pub static SamplePing: Lazy<PingType> =
+        Lazy::new(|| PingType::new("pre-register", true, true, vec![]));
+
+    register_ping_type(&SamplePing);
```

Comment:
For testing purposes we shouldn't need a static, it definitely doesn't need to be `pub`.
`register_ping_type` is also internally called on `PingType::new`, so I _think_ we can change this to just:

```suggestion
    let sample_ping = PingType::new("pre-register", true, true, vec![]);
```